expr/functions/aliasByNode/function_test.go

```diff
@@ -124,14 +131,15 @@ func TestAliasByNode(t *testing.T) {
 			},
 			Want: []*types.MetricData{types.MakeMetricData("base.metric1", []float64{math.NaN(), 1, 1, 1, 1}, 1, now32)},
 		},
-		{
-			Target: "aliasByNode(metric1.fo*.bar.baz,1,3)",
-			M: map[parser.MetricRequest][]*types.MetricData{
-				{Metric: "metric1.fo*.bar.baz", From: 0, Until: 1}: {types.MakeMetricData("metric1.foo==.bar.baz", []float64{1, 2, 3, 4, 5}, 1, now32)},
-			},
-			Want: []*types.MetricData{types.MakeMetricData("foo==.baz",
-				[]float64{1, 2, 3, 4, 5}, 1, now32)},
-		},
+		// FIXME: I don't think `metric1.foo==.bar.baz` is a valid metric name, that is making this test fail https://github.com/grafana/carbonapi/issues/68
```

Comment:
Equal not forbidden in plain metrics. For tags first ident `;` symbols used.

Reply:
You are right. Then I think there's a bug in the `metric.ExtractMetric` helper, because if I test this it fails (here 
https://github.com/go-graphite/carbonapi/blob/b1e09e902ff9b2fd2bc9c8f0ec16d95fe885575d/expr/helper/metric/extract_test.go#L5):
```go
		{
			"metric1.foo.Ab1==.lag",
			"metric1.foo.Ab1==.lag",
		},
```
```
--- FAIL: TestExtractMetric (0.00s)
    --- FAIL: TestExtractMetric/metric1.foo.Ab1==.lag (0.00s)
        /home/nicolas/code/carbonapi/expr/helper/metric/extract_test.go:51: extractMetric("metric1.foo.Ab1==.lag")=".lag", want "metric1.foo.Ab1==.lag"
FAIL
FAIL	github.com/go-graphite/carbonapi/expr/helper/metric	0.002s
FAIL
```

Reply:
Yes, helper.ExtractMetric has this bug. For extatract nodes I use types.ExtractName, but need some checks
https://github.com/go-graphite/carbonapi/blob/main/expr/types/extract.go#L63

Reply:
But it need some adpotation for this

Reply:
I add some fix.
https://github.com/msaf1980/carbonapi/commit/0f10311b44ce040d7f6523fc526b309c8201909c

But now we need fix for all functions. `types.MakeMetricData` don't get manual-added tags (derived from graphite-web).
For example, fixes for transformNull
https://github.com/msaf1980/carbonapi/commit/1e042b829936f00a50b91d088f3447e974726e98

Reply:
> I add some fix.
https://github.com/msaf1980/carbonapi/commit/0f10311b44ce040d7f6523fc526b309c8201909c

Nice, that works perfectly. I'll incorporate the `ExtractNameTag`.

> But now we need fix for all functions for tags check. 

That's another problem, the one we were discussing here: https://github.com/go-graphite/carbonapi/pull/686#discussion_r965673193 . I'm planning to fix all the tests using `types.MakeMetricData` in a separate PR so this one doesn't get bigger. This is the issue for that: https://github.com/grafana/carbonapi/issues/65